ros/lib/utils.py

```diff
@@ -210,46 +210,85 @@ def get_psi_count(queryset: db.Model, bool_flag: bool) -> int:
+    instance_type_properties = None
     if cloud_provider == 'AWS':
-        instance_type_properties = aws_instance_types.INSTANCE_TYPES.get(instance_type, None)
+        instance_type_properties = aws_instance_types.INSTANCE_TYPES.get(
+            instance_type_name, None)
         if instance_type_properties is None:
             # logging lookup failure on Prometheus
             ec2_instance_lookup_failures.labels(reporter='API Events').inc()
-            return 'NA'
+    return instance_type_properties
 
+
+def generate_highlight_description(instance_type, cloud_provider, regions):
+    """Returns string that is description against instance type."""
+    instance_type_properties = instance_type_info_by_name(
+        instance_type, cloud_provider)
+    regions_info = ",".join(regions) + ' regions' if regions else ''
+
+    if instance_type_properties is not None:
         cpu_type = instance_type_properties['extra']['physicalProcessor']
         num_vcpus = instance_type_properties['extra']['vcpu']
         ram_gb = instance_type_properties['extra']['memory']
-        cloud_regions = instance_type_properties['extra']['regionCode']
+        description_text = f'{cpu_type} instance with {num_vcpus} vCPUs ' \
+                           f'and {ram_gb} of RAM, running on {cloud_provider} ' \
+                           f'{regions_info}'
+        return description_text
+
+    return 'NA'
+
+
+def is_current_section(section):
+    """
+        Returns true when section is current
+        otherwise false for rest sections.
+    """
+    return section == 'current'
+
 
-    description_text = f'{cpu_type} instance with {num_vcpus} vCPUs ' \
-                       f'and {ram_gb} of RAM, running on {cloud_provider} ' \
-                       f'{cloud_regions} regions'
+def find_instance_type(section, record):
+    """Get instance type by current, suggested & historical section."""
+    instance_type = None
+    try:
+        if is_current_section(section):
+            instance_type = record.rule_hit_details[0]['details']['instance_type']
+        else:
+            instance_type = record.rule_hit_details[0]['details']['candidates'][0][0]
+    except (IndexError, KeyError):
+        instance_type = None
+
+    if is_current_section(section):
+        return (instance_type if instance_type else record.instance_type)
+
+    return instance_type
 
-    return description_text
+
+def find_candidates_and_regions(highlight_type, dataset):
+    """Find candidates by highlight_type & regions per candidate ."""
+    instance_candidates, regions = [], {}
+    for record in dataset:
+        instance_type = find_instance_type(highlight_type, record)
+        if instance_type is None:
+            continue
+        instance_candidates.append(instance_type)
+        if instance_type not in regions:
+            regions[instance_type] = []
+        if record.region not in regions[instance_type]:
+            regions[instance_type].append(record.region)
```

Comment:
On line 277 making `regions[instance_type]` as an array will create an issue. Doing this will print all the regions in description column of Executive report.  
Consider the scenario where user has 2 systems of same type(t2.micro) both in different region(ap-south-1 and us-east-1) in this case if user will generate executive report he will see below.

| instance type | # of systems | Description                                                                                        |
|---------------|--------------|----------------------------------------------------------------------------------------------------|
| t2.micro      | 1            | AMD EPYC 7571 instance with 2 vCPUs and 8 GiB of RAM, running on AWS ap-south-1, us-east-1 regions |
| t2.micro      | 1            | AMD EPYC 7571 instance with 2 vCPUs and 8 GiB of RAM, running on AWS ap-south-1, us-east-1 regions |

Both row will have count 1 but if you see the description it shows two regions.

Reply:
Oh my bad, Please ignore the above. I didn't notice before but [line 291](https://github.com/RedHatInsights/ros-backend/pull/340/files#diff-73b796983793ca68d788f97a84b00ef19df5157d5fee7ee854c47ed1133d45f1R291) will take care of grouping the instance count.